Sources/SwiftSyntaxMacros/MacroProtocols/BodyMacro.swift

```diff
@@ -29,4 +29,31 @@ public protocol BodyMacro: AttachedMacro {
     providingBodyFor declaration: some DeclSyntaxProtocol & WithOptionalCodeBlockSyntax,
     in context: some MacroExpansionContext
   ) throws -> [CodeBlockItemSyntax]
+
+  /// Expand a macro described by the given custom attribute and
+  /// attached to the given closure and evaluated within a
+  /// particular expansion context.
+  ///
+  /// The macro expansion can replace the body of the given closure.
+  static func expansion(
+    of node: AttributeSyntax,
+    providingBodyFor closure: ClosureExprSyntax,
+    in context: some MacroExpansionContext
+  ) throws -> [CodeBlockItemSyntax]
+}
+
+private struct ClosureNotSupported: Error, CustomStringConvertible {
+  var description: String {
+    "Function body macro cannot be applied to closure"
+  }
+}
+
+extension BodyMacro {
+  public static func expansion(
```

Comment:
Should we mark this default implementation as deprecated to prompt clients to add it?

Reply:
It's reasonable for a function body macro to support function declarations and not closures, but yeah that should probably be a deliberate decision by the macro author by implementing the requirement explicitly rather than the default behavior.

Reply:
I'll wait to add the deprecation until the RFC and SE pitch discussion in case the requirement ends up being tweaked in some way.